RevenueCatUI/Templates/Components/Image/ImageComponentViewModel.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ImageComponentView.swift
+//
+//  Created by Josh Holtz on 6/11/24.
+// swiftlint:disable missing_docs
+
+import Combine
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+public class ImageComponentViewModel: ObservableObject {
+
+    let locale: Locale
+    @Published private(set) var component: PaywallComponent.ImageComponent
```

Comment:
I thought I asked this the first time but I can't remember 😅 

What's the benefit of this being an `ObservableObject` with a `@Published` property? 🤔 If we process are everything upfront, this view model _shouldn't_ change at all, right?

Reply:
I don't think we discussed this - I wrote it like that because I wanted to create a live paywall (for my loom and for customers in the future) where it would fetch the model every second or so, and have the view update automatically when the model changed. Somewhere this wasn't working and rather than spend more time debugging it I moved no, but left this as-is for the future. If we removed it now it would have no effect, so I guess maybe we should until if/when functionality is available.

Reply:
> I wrote it like that because I wanted to create a live paywall

Oh neat! That sounds kind of a neat way to debug/test some designs with the dashboard 😁 

> If we removed it now it would have no effect, so I guess maybe we should until if/when functionality is available.

I think it _might_ be best to remove for now if easy enough! Just to make the state/flow a little easier with these being immutable

Reply:
Done! https://github.com/RevenueCat/purchases-ios/pull/4230/commits/56b52cb276cc627d43a913aecfa013fc58ee8c2c